tests/phpunit/Civi/API/RequestTest.php

```diff
@@ -68,4 +68,34 @@ public function testCreateRequest_InvalidEntityAction($inEntity, $inAction, $inV
     Request::create($inEntity, $inAction, ['version' => $inVersion], NULL);
   }
 
+  public function getCastingExamples(): array {
+    $exs = [];
+    // We run the same tests on `$checkPermissions` (which has real PHP setter method)
+    // and `$useTrash` (which has a generic magic method) to show that casting is similar.
+    $exs[] = ['Contact.delete checkPermissions', 0, FALSE];
+    $exs[] = ['Contact.delete checkPermissions', '0', FALSE];
+    $exs[] = ['Contact.delete checkPermissions', 1, TRUE];
+    $exs[] = ['Contact.delete checkPermissions', '1', TRUE];
+    $exs[] = ['Contact.delete useTrash', 0, FALSE];
+    $exs[] = ['Contact.delete useTrash', '0', FALSE];
+    $exs[] = ['Contact.delete useTrash', 1, TRUE];
+    $exs[] = ['Contact.delete useTrash', '1', TRUE];
+    return $exs;
+  }
+
+  /**
+   * @param $entityActionField
+   * @param $inputValue
+   * @param $expectValue
```

Comment:
`bool`?

Reply:
The test is written in such a way that it _could_ have non-boolean expected values. At the moment all the expected values are bools. Maybe it would be good to put in at least one non-boolean, like `$exs[] = ['Contact.get limit', 1, 1];`

Reply:
Yeah, this is right - `mixed` is the best representation, since it could handle other fields/types.

I've updated the test a fair bit -- to declare `@params` better, to improve test-performance, and to cover a couple other data-types that are fairly straight-forward (`int`, `float`).